app/tournament/tournament.js

```diff
@@ -0,0 +1,101 @@
+import * as player from "./player";
+import * as match from "./match";
+import * as utils from "../common/utils";
+
+export class Tournament {
+    constructor() {
+        this.players = [];
+        this.matches = [];
+    }
+
+    addPlayer(player) {
+        if (this.existsAlready(player)) {
+            throw new TournamentError(`Duplicate player: "${player.name}".`);
```

Comment:
No, you can keep the throw, it´s just that your logic must avoid it. If it happens, you have a bug, so you don´t even need to catch it, you have to fix the bug.

It´s like a/b, I don´t want to catch the DivisionByZero error, I just want to ensure that everytime I make to that line, b is not zero. If it is, I have a bug.